extractors/imagingextractor.py

```diff
@@ -0,0 +1,86 @@
+from abc import ABC, abstractmethod
+import numpy as np
+from spikeextractors.baseextractor import BaseExtractor
+from .extraction_tools import ArrayType, PathType, NumpyArray, DtypeType
+
+
+class ImagingExtractor(ABC, BaseExtractor):
+    '''An abstract class that contains all the meta-data and input data from
+       the imaging data.
+    '''
+    def __init__(self):
+        BaseExtractor.__init__(self)
+        self._memmapped = False
+
+    @abstractmethod
+    def get_frame(self, frame_idx) -> NumpyArray:
+        pass
+
+    @abstractmethod
+    def get_frames(self, frame_idxs) -> NumpyArray:
+        pass
+
+    @abstractmethod
+    def get_video(self, start_frame=None, end_frame=None) -> NumpyArray:
+        pass
+
+    @abstractmethod
+    def get_image_size(self) -> ArrayType:
+        pass
+
+    @abstractmethod
+    def get_num_frames(self) -> int:
+        pass
+
+    @abstractmethod
+    def get_sampling_frequency(self) -> float:
+        pass
+
+    @abstractmethod
+    def get_dtype(self) -> DtypeType:
+        pass
```

Comment:
It seems like this could be defined in the abstract class

Reply:
with a `return self.get_frame(0).dtype`?